### PR TITLE
clone config

### PR DESCRIPTION
**Pull request by Aaron2000119**
Part of https://github.com/Gamify-IT/issues/issues/387

Adds the option to clone a configuration.

To test use the description in the overworld-backend pull request.

### CODE REVIEW

File: src/main/java/de/unistuttgart/bugfinder/configuration/ConfigurationController.java [hunk 1]
```diff
@@ -67,4 +67,10 @@ public ConfigurationDTO deleteConfiguration(@PathVariable final UUID id) {
     log.debug("DELETE /configurations/{}", id);
     return configurationService.delete(id);
   }
+
+  @PostMapping("/configurations/{id}/clone")
+  public UUID cloneConfiguration(@PathVariable final UUID id) {
+    log.debug("Clone /configurations/{}", id);
+    return configurationService.cloneConfiguration(id);
```
Comment (delvh) [inline src/main/java/de/unistuttgart/bugfinder/configuration/ConfigurationController.java:100]: No `is lecturer` check?

File: src/main/java/de/unistuttgart/bugfinder/configuration/ConfigurationService.java [hunk 2]
```diff
@@ -150,6 +150,12 @@ public ConfigurationDTO delete(final UUID id) {
     Configuration configuration = configurationRepository
       .findById(id)
       .orElseThrow(() -> new ResponseStatusException(HttpStatus.NO_CONTENT));
+    for (Code code : configuration.getCodes()) {
+      if (solutionRepository.findByCodeId(code.getId()).isPresent()) {
+        Solution solution = solutionRepository.findByCodeId(code.getId()).get();
+        solutionRepository.deleteById(solution.getId());
+      }
```
Comment (delvh) [inline src/main/java/de/unistuttgart/bugfinder/configuration/ConfigurationService.java]:
```suggestion
      solutionRepository.findByCodeId(code.getId()).ifPresent(solution -> solutionRepository.deleteById(solution.getId()));
      }
```

File: src/main/java/de/unistuttgart/bugfinder/configuration/ConfigurationService.java [hunk 3]
```diff
@@ -150,6 +150,12 @@ public ConfigurationDTO delete(final UUID id) {
...
+    }
```
Comment (delvh) [inline src/main/java/de/unistuttgart/bugfinder/configuration/ConfigurationService.java]:
```suggestion
      configuration.getCodes().parallelStream().map(code -> solutionRepository.findByCodeId(code.getId())).filter(Optional::isPresent).forEach(solution -> solutionRepository.deleteById(solution.getId()));
```

**Review by maexled — APPROVED** [review]
lgtm

**Pull request merged**